auslib/test/blobs/test_systemaddons.py

```diff
@@ -222,3 +222,26 @@ def testDoesNotContainForbiddenDomain(self):
 """)
         self.assertFalse(blob.containsForbiddenDomain('gg',
                                                       self.whitelistedDomains))
+
+    def testIncorrectAddonLayout(self):
+        blob = SystemAddonsBlob()
+        blob.loadJSON("""
+    {
+    "name": "fake",
+    "schema_version": 1000,
+    "hashFunction": "SHA512",
+    "addons": {
+        "c": {
+            "version": "1",
+            "platforms": {
+                "p": {
+                    "filesize": 2,
+                    "hashValue": "3",
+                    "fileUrl": "http://a.com/blah"
+                }
+            }
+        }
+    }
+}
+""")
+        blob.validate('gg', self.whitelistedDomains)
```

Comment:
Don't go by the function_name, the blob inside is actually valid. I wrote this test_name to check for invalid blobs only but temporarily, it contains a valid blob for verification. (Sorry for this)